Magento_Theme/web/js/view/movies/movie-modal.js

```diff
@@ -0,0 +1,114 @@
+define(['uiElement', 'jquery'], (uiElement, $) => {
+  return uiElement.extend({
+    defaults: {
+      template: 'Magento_Theme/movies/movie-modal-layout',
+      links: {
+        movieList: '${ $.parentName }:movieList',
+        movie: '${ $.parentName }:movie',
+        modalMode: '${ $.parentName }:modalMode',
+      },
+      tracks: {
+        movieList: true,
+        movie: true,
+      },
+      modules: {
+        parentComponent: '${ $.parentName }',
+      },
+      listens: {
+        movie: 'handleMovieInputFields',
+      },
+    },
+
+    initialize() {
+      this._super();
+    },
+
+    randomIdNumber() {
+      const randomNumber = Math.floor(Math.random() * 100000).toString();
+
+      return 'tt00' + randomNumber.padStart(5, '0');
+    },
+
+    clearInputFields() {
+      $('#title').val('');
+      $('#year').val('');
+      $('#poster').val('');
+    },
+
+    closeMovieModal(e) {
+      e.preventDefault();
+      $('.movie-modal').hide();
+    },
+
+    handleMovieInputFields() {
+      setTimeout(() => {
+        if (this.modalMode === 'edit' && this.movie) {
+          $('#title').val(this.movie.Title);
+          $('#year').val(this.movie.Year);
+          $('#poster').val(this.movie.Poster);
+          $('.movie-modal__btn-add').text('Edit');
+        } else {
+          this.clearInputFields();
+          $('.movie-modal__btn-add').text('Add');
+        }
+      }, 200);
+    },
+
+    editExistingMovie(e) {
+      e.preventDefault();
+
+      const title = $('#title').val();
+      const year = $('#year').val();
+      const poster = $('#poster').val();
+
+      if (this.modalMode === 'edit' && this.movie) {
```

Comment:
use earlier exit to avoid nesting
```javascript
if (this.modalMode !== 'edit' || !this.movie) {
 return this;
}

// next code on the same level

if (movieIndex === -1) {
 return this;
}

// next code on the same level
```